app/jobs/shipit/merge_pull_requests_job.rb

```diff
@@ -13,7 +13,14 @@ def perform(stack)
       return false unless stack.allows_merges?
 
       pull_requests.select(&:pending?).each do |pull_request|
-        return false unless pull_request.merge!
+        # Any merged PR might conflict with others, so we need to refresh.
+        # If the PR is unchanged, we should get a cache hit anyway.
+        pull_request.refresh!
+        unless pull_request.merge!
+          # That PR isn't mergeable yet, we'll try again a bit later.
+          MergePullRequestsJob.set(wait: 10.seconds).perform_later(stack)
```

Comment:
Are we risking an endless loop here if the PR that got merged now legitimately has merge conflicts? Or does the "refresh" make sure it is no longer pending it that case?